islandCountTest.java

```diff
@@ -0,0 +1,66 @@
+import static org.junit.Assert.*;
+
+import org.junit.Test;
+
+public class islandCountTest {
+
+	// Set up of map
+	public Boolean[][] mapSetUp() {
+		
+		Boolean map[][] = new Boolean[4][4];
+		
+		map[0][0] = false;
+		map[0][1] = true;
+		map[0][2] = false;
+		map[0][3] = true;
+		
+		map[1][0] = true;
+		map[1][1] = true;
+		map[1][2] = false;
+		map[1][3] = false;
+		
+		map[2][0] = false;
+		map[2][1] = false;
+		map[2][2] = true;
+		map[2][3] = false;
+		
+		map[3][0] = false;
+		map[3][1] = false;
+		map[3][2] = true;
+		map[3][3] = false;
+		
+		// F T F T
+		// T T F F
+		// F F T F
+		// F F T F
+		
+		return map;
+	}
+	
+	// Test correct number of trues become false
+	@Test
+	public void testSameIsland() {
+		
+		Boolean map[][] = mapSetUp();
+		Islands world = new Islands();
+		
+		assertEquals(3, world.sameIslandCount(map, 0, 1));
```

Comment:
Calling world.sameIslandCount(map, 0, 1) then world.sameIslandCount(map, 1, 0) would result in an error because map has been modified. The caller wouldn't expect this to happen.